### ISSUE TITLE
save_hyperparameters() saves all **kwargs arguments - which includes unpickable plugins for DDP

### ISSUE THREAD

**Issue by Renthal** [PyTorchLightning/pytorch-lightning#9984]
## 🐛 Bug

The call to save `self.save_hyperparameters()` (NOTE: no args provided) as described in the docs (see [here](https://pytorch-lightning.readthedocs.io/en/latest/common/hyperparameters.html#lightningmodule-hyperparameters
)) is by default saving the content of **kwargs to hparams. This is an issue because:

1. this is not matching the documentation (see the 'equivalent' call part)
2. one of the fields (I suspect introduced between 1.2.8 to 1.4.9?) is 'plugins' and there there is a DDP related object which cannot be pickled. This causes model checkpoints saved like this to be impossible to be loaded outside of a similar env (boh with PL loading functionality or the torch.load method).

### To Reproduce

Save any model according to the docs and try to load it in another script which runs afterwards (i.e. in a inference only script) not powered by PL

### Expected behavior

It should keep working as it did in version 1.2.8

### Environment

- PyTorch Lightning Version 1.4.9

**Comment by awaelchli** [comment]
@Renthal I may not fully understand the issue you are describing, but are you saying that you pass in some arguments from the trainer to the `save_hyperparameters()` method? Our documentation shows an example yes, but I would strongly advise against that, as these should not be considered hyperparameters to the model. Could you share which plugin it is that causes trouble?

**Comment by Renthal** [comment]
I can add some context yes. So, if I run the code with this code
```
 def __init__(self, **kwargs):
        self.save_hyperparameters() 
```
all the stuff inside kwargs gets put into the `self.hparams` field, which ins 1.2.8 was OK and quite convenient. I then updated to 1.4.9 and "all of a sudden" I could not load a checkpoint anymore for some very cryptic error message about DDP (note: I was loading the checkpoint for inference on cpu-only, so I found it very weird). A lot of digging brought me to find out that the field `self.hparams.plugins' contains something DDP-related and this was not pickable, making the checkpoint unusable in my context. Further, I figured that the field `logger` was also being stored there  which create a import error (I am using wandb) in the inference environment where I don't have that logger installed. 
So the final version to make it work for me was:
```
 def __init__(self, **kwargs):
        self.save_hyperparameters(ignore=['logger', 'profiler', 'plugins']) 
```
The weird part is that I changed nothing in my code but the PL version from 1.2.8 to 1.4.9 so assume (but maybe I am wrong?) that the change was internal. Further, if following the documentation the code does not work I have a hard-time to debug. It must be noted that if you load a checkpoint in the PL env it might as well work (since DDP is setup) but then this makes the checkpoints not portable and this is definitely not desirable. I hope this help explaining the issue.
